Mathlib/Tactic/Linter/PPRoundtrip.lean

```diff
@@ -60,7 +60,7 @@ def polishSource (s : String) : String × Array Nat :=
   let preWS := split.foldl (init := #[]) fun p q =>
     let txt := q.trimLeft.length
     (p.push (q.length - txt)).push txt
-  let preWS := preWS.eraseIdx 0
+  let preWS := preWS.eraseIdx! 0
```

Comment:
Does `eraseIdx!` panic?

Reply:
Which is to say; I think this might have been relying on the old behavior that `eraseIdx` was a no-op if the index was out of bounds.

Reply:
Okay, I'll change this to `eraseIdxIfInBounds`. It would be lovely if we could have a test demonstrating that it is relying on this, I guess. @adomani, do you know if what this should do?

Reply:
I am late for this party!

Anyway, `preWS` is always (I hope!) an array with at least 2 entries: `split` is obtained via `String.split` on some string and therefore always consists of at least one entry.  Next, `preWS` loops over the entries of `split` appending two natural numbers on every loop.  The result is `preWS` that looks like
```lean
#[0, <an odd number of remaining entries>]
```
So, whether or not you use a function that panics there, I think that the resulting code will not panic.